src/main/java/org/opensearch/ad/transport/ValidateAnomalyDetectorResponse.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.transport;
+
+import java.io.IOException;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.ad.model.DetectorValidationIssue;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+public class ValidateAnomalyDetectorResponse extends ActionResponse implements ToXContentObject {
+    private DetectorValidationIssue issue;
+
+    public DetectorValidationIssue getIssue() {
+        return issue;
+    }
+
+    public ValidateAnomalyDetectorResponse(DetectorValidationIssue issue) {
+        this.issue = issue;
+    }
+
+    public ValidateAnomalyDetectorResponse(StreamInput in) throws IOException {
+        super(in);
+        if (in.available() > 0) {
+            issue = new DetectorValidationIssue(in);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (issue != null) {
```

Comment:
How about we write false to output stream if issue is null? So we can check if `in.readBoolean()` is true in line 37.
If we need to add more fields to this class in future, for example we add filed "suggestion",  if issue is null , suggestion not null, will be hard to know if the value read from input stream is for issue or for suggestion.

Reply:
added